internal/cmd/stack/run_logs.go

```diff
@@ -205,7 +205,7 @@ func runStateLogs(ctx context.Context, stack, run string, state structs.RunState
 			sink <- message.Body
 		}
 
-		if logs.Finished {
+		if logs.Finished || !logs.HasMore && stateTerminal {
```

Comment:
```suggestion
		if logs.Finished || (!logs.HasMore && stateTerminal) {
```
Is this the same? So I don't have to think about operator priority.